Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -1777,6 +2124,594 @@ RefPtr<StyleValue> Parser::parse_box_shadow_value(ParsingContext const& context,
+                        return nullptr;
+                    maybe_line_height = line_height.release_nonnull();
+                    i += 2;
+                }
+            }
+
+            // Consume font-families
+            auto font_families = parse_font_family_value(context, component_values, i + 1);
+            if (!font_families)
+                return nullptr;
+            maybe_font_families = font_families.release_nonnull();
+            break;
+        }
+
+        return nullptr;
+    }
+
+    // Since normal is the default value for all the properties that can have it, we don't have to actually
+    // set anything to normal here. It'll be set when we create the FontStyleValue below.
+    // We just need to make sure we were not given more normals than will fit.
+    int unset_value_count = (maybe_font_style.has_value() ? 1 : 0) + (maybe_font_weight.has_value() ? 1 : 0);
+    if (unset_value_count < normal_count)
+        return nullptr;
+
+    if (!maybe_font_size.has_value() || !maybe_font_families.has_value())
+        return nullptr;
+
+    auto normal = IdentifierStyleValue::create(ValueID::Normal);
+    return FontStyleValue::create(
+        maybe_font_style.value_or(normal),
+        maybe_font_weight.value_or(normal),
+        maybe_font_size.value(),
+        maybe_line_height.value_or(normal),
+        maybe_font_families.value());
+}
+
+RefPtr<StyleValue> Parser::parse_font_family_value(ParsingContext const& context, Vector<StyleComponentValueRule> const& component_values, size_t start_index)
+{
+    auto is_generic_font_family = [](ValueID identifier) -> bool {
+        switch (identifier) {
+        case ValueID::Cursive:
+        case ValueID::Fantasy:
+        case ValueID::Monospace:
+        case ValueID::Serif:
+        case ValueID::SansSerif:
+        case ValueID::UiMonospace:
+        case ValueID::UiRounded:
+        case ValueID::UiSerif:
+        case ValueID::UiSansSerif:
+            return true;
+        default:
+            return false;
+        }
+    };
+
+    auto is_comma_or_eof = [&](size_t i) -> bool {
+        if (i < component_values.size()) {
+            auto& maybe_comma = component_values[i];
+            if (!maybe_comma.is(Token::Type::Comma))
+                return false;
+        }
+        return true;
+    };
+
+    // Note: Font-family names can either be a quoted string, or a keyword, or a series of custom-idents.
+    // eg, these are equivalent:
+    //     font-family: my cool     font\!, serif;
+    //     font-family: "my cool font!", serif;
+    NonnullRefPtrVector<StyleValue> font_families;
+    Vector<String> current_name_parts;
+    for (size_t i = start_index; i < component_values.size(); ++i) {
+        auto& part = component_values[i];
+
+        if (part.is(Token::Type::String)) {
+            // `font-family: my cool "font";` is invalid.
+            if (!current_name_parts.is_empty())
+                return {};
```

Comment:
`return nullptr` is what you do elsewhere. I think `RefPtr`s are defaulted to `nullptr`, so functionally there's no difference.
Same for all `return {}` in this function.

Reply:
Good catch. The `parse_font_family_value()` changed quite a lot as I was working on it, and at one point it returned Optional. But since I'm using nullptr for everywhere else I'll change it to use that here too.